component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/duo/DuoAuthenticator.java

```diff
@@ -76,13 +77,25 @@ protected void initiateAuthenticationRequest(HttpServletRequest request, HttpSer
             throws AuthenticationFailedException {
 
         String username;
+        AuthenticatedUser authenticatedUser;
         Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
         String integrationSecretKey = DuoAuthenticatorConstants.stringGenerator();
         context.setProperty(DuoAuthenticatorConstants.INTEGRATION_SECRET_KEY, integrationSecretKey);
         context.setProperty(DuoAuthenticatorConstants.AUTHENTICATION, DuoAuthenticatorConstants.AUTHENTICATOR_NAME);
         FederatedAuthenticatorUtil.setUsernameFromFirstStep(context);
         username = String.valueOf(context.getProperty(DuoAuthenticatorConstants.DUO_USERNAME));
 
+        authenticatedUser = (AuthenticatedUser) context.getProperty(DuoAuthenticatorConstants.AUTHENTICATED_USER);
+        if (authenticatedUser != null) {
```

Comment:
What happens when the authenticatedUser is null?

Reply:
if the authenticated user is null then we can't consider the configuration since we don't have tenant domain, user store domain. Ideally authenticated user will be not be null since duo will be configured with second step always.